### PR TITLE
fix: Prevent simultaneous balance of segments and channels

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #33550
balance segment and balance segment execute at same time, which will cause bounch of corner case.

This PR disable simultaneous balance of segments and channels

### CODE REVIEW

**Comment by weiliu1031** [comment]
rerun go-sdk

**Comment by weiliu1031** [comment]
rerun ut

**Comment by weiliu1031** [comment]
rerun go-sdk

**Comment by weiliu1031** [comment]
rerun ut

**Review by tedxu — COMMENTED** [review]
I agree we should address the conflicts in some extent, e.g., moving the same segment from and to the same node should be avoided. However, I'm not sure if completely disabling the concurrency of a collection is a good idea. Could this change significantly slow down the balance process (say there is cluster reconfigure/scale out)?

File: internal/querycoordv2/balance/balance.go [hunk 1]
```diff
@@ -141,6 +141,14 @@ func (b *RoundRobinBalancer) BalanceReplica(replica *meta.Replica) ([]SegmentAss
 	return nil, nil
 }
 
+func (b *RoundRobinBalancer) permitBalanceChannel(collectionID int64) bool {
+	return b.scheduler.GetSegmentTaskNum(task.WithCollectionID2TaskFilter(collectionID), task.WithTaskTypeFilter(task.TaskTypeMove)) == 0
```
Comment (czs007) [inline internal/querycoordv2/balance/balance.go:145]: how about making this zero configurable?

**Comment by tedxu** [comment]
Resolved my concern per offline discussion with @weiliu1031 , it /lgtm now.

Still, the segment-channel exclusion can be narrowed down to a specific channel rather than to the whole collection, it can be improved in another PR.

**Pull request merged**